src/tiled/variantpropertymanager.cpp

```diff
@@ -126,6 +135,32 @@ QString VariantPropertyManager::valueText(const QtProperty *property) const
         QVariant value = mValues[property].value;
         int typeId = propertyType(property);
 
+        if (typeId == objectIdTypeId()) {
+            ObjectId id = value.value<ObjectId>();
+
+            if (MapDocument *document = DocumentManager::instance()->currentDocument()) {
```

Comment:
In the TMX file there should always just be the ID. The `Map*` and `Tile*` are only needed for the `VariantPropertyManager` to have enough context information to display the name of the object. In the TMX file, this context is clear.